src/venv_manager.rs

```diff
@@ -250,7 +257,58 @@ impl VenvManager {
         Ok(())
     }
 
-    fn run_pip_freeze(&self) -> Result<(), Error> {
+    fn write_lock(&self) -> Result<(), Error> {
+        let metadata = &self.get_metadata()?;
+
+        let lock_path = &self.paths.lock;
+        let lock_contents = if lock_path.exists() {
+            std::fs::read_to_string(&lock_path).map_err(|e| Error::ReadError {
+                path: lock_path.to_path_buf(),
+                io_error: e,
+            })?
+        } else {
+            String::new()
+        };
+
+        let mut lock = Lock::from_string(&lock_contents)?;
+        let frozen_deps = self.get_frozen_deps()?;
+        lock.freeze(&frozen_deps);
+        let new_contents = lock.to_string();
+
+        let LockMetadata {
+            dmenv_version,
+            python_version,
+            python_platform,
+        } = metadata;
+        let top_comment = format!(
+            "# Generated with dmenv {}, python {}, on {}\n",
+            dmenv_version, &python_version, &python_platform
+        );
+
+        let to_write = top_comment + &new_contents;
+        std::fs::write(&lock_path, &to_write).map_err(|e| Error::WriteError {
+            path: lock_path.to_path_buf(),
+            io_error: e,
+        })?;
+        Ok(())
```

Comment:
You can directly remove the fs::write's result in this case, no need for try and Ok(())